src/main/java/com/mineboundteam/minebound/entity/FireProjectile.java

```diff
@@ -0,0 +1,124 @@
+package com.mineboundteam.minebound.entity;
+
+import com.mineboundteam.minebound.entity.registry.EntityRegistry;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.particles.ParticleOptions;
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.projectile.Fireball;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.BaseFireBlock;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.EntityHitResult;
+import net.minecraft.world.phys.HitResult;
+import org.jetbrains.annotations.NotNull;
+
+public class FireProjectile extends Fireball {
```

Comment:
Is there a reason we need a custom class instead of just using Fireball?

Reply:
Can't modify the damage, whether or not it ignites blocks, or the maximum distance of a `Fireball`.

Reply:
Oh, interesting